crates/view/src/planner.rs

```diff
@@ -222,9 +222,33 @@ impl<R: RngCore + CryptoRng> Planner<R> {
     }
 
     /// Withdraws the reserves of the Dutch auction.
-    // TODO: nicer api? what do we get by passing fields individually rather than the plan?
+    ///
+    /// Uses the provided auction state to automatically end the auction
+    /// if necessary.
     #[instrument(skip(self))]
-    pub fn dutch_auction_withdraw(&mut self, plan: ActionDutchAuctionWithdrawPlan) -> &mut Self {
+    pub fn dutch_auction_withdraw(&mut self, auction: &DutchAuction) -> &mut Self {
+        let auction_id = auction.description.id();
+        // Check if the auction needs to be ended
+        if auction.state.sequence == 0 {
+            self.dutch_auction_end(auction_id);
+        }
+
+        let reserves_input = Value {
+            amount: auction.state.input_reserves,
+            asset_id: auction.description.input.asset_id,
+        };
+        let reserves_output = Value {
+            amount: auction.state.output_reserves,
+            asset_id: auction.description.output_id,
+        };
+
+        let plan = ActionDutchAuctionWithdrawPlan {
+            auction_id,
+            seq: 2,
```

Comment:
is this intentionally hardcoded to `2`? it would be nice to leave a comment explaining why this is `2`, if so

Reply:
```suggestion
            seq: 2, // 1 (closed) -> 2 (withdrawn)
```

good point, here the sequence number corresponds to a withdrawn auction. It's documented in [`DutchAuction`] but added another comment to re-anchor it